packages/plugin-toolbar/src/components/ToolbarComponent.ts

```diff
@@ -0,0 +1,45 @@
+import { OwlUIComponent } from '../../../owl-ui/src/OwlUIComponent';
+import { CommandIdentifier, CommandParams } from '../../../core/src/Dispatcher';
+import { Toolbar, ToolbarLayout, Button, ToolbarItem } from '../Toolbar';
+
+export class ToolbarComponent extends OwlUIComponent<{}> {
+    static components = {};
+    static template = 'toolbar';
+    config: ToolbarLayout = this.env.editor.plugins.get(Toolbar).configuration.layout || [];
+
+    async willStart(): Promise<void> {
+        this.env.editor.dispatcher.registerCommandHook('*', this.refresh.bind(this));
+        return super.willStart();
+    }
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    execCommand(commandId: CommandIdentifier, commandArgs?: CommandParams): void {
+        this.env.editor.execCommand(commandId, commandArgs);
+    }
+    isEnabled(item: Button): boolean {
+        return !item.disabled || !item.disabled(this.env.editor);
+    }
+    isSelected(item: Button): boolean {
+        return item.pressed && item.pressed(this.env.editor);
+    }
+    isArray(item: ToolbarItem): boolean {
+        return Array.isArray(item);
+    }
+    isString(item: ToolbarItem): boolean {
+        return typeof item === 'string';
+    }
+    onDropdownChange(dropdown: Button, event: Event): void {
+        const selectedIndex = (event.target as HTMLSelectElement).selectedIndex;
+        // -1 to account for the mandatory empty option on top:
+        const dropdownItem = dropdown[selectedIndex - 1];
```

Comment:
Shouldn't you account for strings in the dropdown as well ? Would you need to `as` the event if you chose the right type for `event` in the function signature ?

Reply:
What do you mean with the strings in the dropdown? If you mean the strings in the configuration, they do become `<option>`s themselves, just disabled, so they count for the index.
The event is just a change event so I would still need the `as` given that it has no idea the change is a `<select>` change.